core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/BaseEventStreamAsyncAws4Signer.java

```diff
@@ -259,4 +267,35 @@ public void subscribe(Subscriber<? super ByteBuffer> s) {
             return transformedRequestBody.contentLength();
         }
     }
+
+    private static String toDebugString(Message m, boolean truncatePayload) {
```

Comment:
I know we don't usually write unit tests for logs but maybe this method is complex enough to justify one?

Reply:
Sure will add one